src/test/java/seedu/internhunter/logic/parser/delete/DeleteCommandParserTest.java

```diff
@@ -15,12 +26,70 @@
  */
 public class DeleteCommandParserTest {
 
-    private DeleteCommandParser parser = new DeleteCommandParser();
+    private static final String VALID_COMPANY_INPUT = " com 1";
+    private static final String VALID_INTERNSHIP_INPUT = " int 1 i/1";
+    private static final String VALID_APPLICATION_INPUT = " app 1";
+    private static final String VALID_PROFILE_INPUT = " me 1";
+    private DeleteCommandParser deleteCommandParser = new DeleteCommandParser();
+
+    @BeforeEach
+    public void setUp() {
+        deleteCommandParser = new DeleteCommandParser();
+    }
 
     @Test
-    public void parse_invalidArgs_throwsParseException() {
-        // todo: change the test case to use the currently commented-out line when all 3 delete command parsers for
-        //  individual items have been implemented
-        assertParseFailure(parser, "a", Messages.MESSAGE_INVALID_ITEM_TYPE);
+    public void parse_invalidTypes_throwsParseException() {
+        // invalid item type
+        assertThrows(ParseException.class, () -> deleteCommandParser.parse(" hello"));
+        assertThrows(ParseException.class, () -> deleteCommandParser.parse(" 1"));
+        assertThrows(ParseException.class, () -> deleteCommandParser.parse(" "));
     }
+
+    @Test
+    public void parse_missingInput_throwsParseException() {
+        // missing description as input
+        assertThrows(ParseException.class, () -> deleteCommandParser.parse(COMPANY_ALIAS));
+        assertThrows(ParseException.class, () -> deleteCommandParser.parse(COMPANY_ALIAS + " "));
+
+        assertThrows(ParseException.class, () -> deleteCommandParser.parse(INTERNSHIP_ALIAS));
+        assertThrows(ParseException.class, () -> deleteCommandParser.parse(INTERNSHIP_ALIAS + " "));
+
+        assertThrows(ParseException.class, () -> deleteCommandParser.parse(APPLICATION_ALIAS));
+        assertThrows(ParseException.class, () -> deleteCommandParser.parse(APPLICATION_ALIAS + " "));
+
+        assertThrows(ParseException.class, () -> deleteCommandParser.parse(PROFILE_ALIAS));
+        assertThrows(ParseException.class, () -> deleteCommandParser.parse(PROFILE_ALIAS + " "));
+    }
+
+    @Test
+    public void parse_addCommandParserPassingToCorrectParser_true() throws ParseException {
```

Comment:
should this and the method below have the name `deleteCommand...` rather than `addCommand...`?

Reply:
fixed